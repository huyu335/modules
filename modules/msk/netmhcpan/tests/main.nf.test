nextflow_process {

    name "Test Process NETMHCPAN"
    script "../main.nf"
    process "NETMHCPAN"

    tag "modules"
    tag "modules_nfcore"
    tag "netmhcpan"

    // TODO nf-core: Change the test name preferably indicating the test-data and file-format used
    test("test_neoantigen") {

        when {
            process {
                """                
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.test_data_mskcc['neoantigen']['temp_test_short_maf'], checkIfExists: true),
                    ]

                input[1] = [
                    file(params.test_data_mskcc['neoantigen']['winners_hla_txt'], checkIfExists: true),
                    ]

                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.versions,
                    process.out.mutatedSequences).match()
                },
                { assert process.out.xls[0][1] ==~ ".*/test.xls" },
                { assert process.out.netmhcpanoutput[0][1] ==~ ".*/test.netmhcpan.output" }
            )
        }

    }

}
