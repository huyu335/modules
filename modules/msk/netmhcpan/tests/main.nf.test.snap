{
    "test_neoantigen": {
        "content": [
            [
                "versions.yml:md5,35ec563839ee27410cf9f8d134c6e8e5"
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.mutated_sequences.fa:md5,74efe5d1fdca3c02b07704515c47e7dd",
                    "test.WT_sequences.fa:md5,9743be25e7cffcf726f51567340e1f25"
                ]
            ],
            "test.MUT.xls",
            "test.WT.xls",
            "test.MUT.netmhcpan.output",
            "test.WT.netmhcpan.output"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.0"
        },
        "timestamp": "2024-03-14T15:45:55.929566"
    }
}